### DISCUSSION TITLE
Cant seem to get Field probe (Reduced diagnostics) running

### DISCUSSION THREAD

**Discussion by sheroytata** [BLAST-WarpX/warpx#4502] [Data Analysis] [answered]
Reduced diagnostics:

Cant seem to get a field probe running in a moving window in the Z direction.

it dumps only the 0th step with no updates to the data as the simulation runs. As I understand, the files should be updated every 10 steps with the iteration/time[[0]step()/[1]time(s)] column modifying in the data file every time data is dumped.

All I get is 0's (since laser does not exist at time step 0) and a linear z, as the field probe setup.

The probe is defined within the initial window. And as I understand the probe should move automatically with the v_window.

Tried also with WarpX@23.08, 23.05
Setup:

```
# Reduced Diagnostics
warpx.reduced_diags_names = EM_FP

EM_FP.type = FieldProbe

EM_FP.intervals = 10
EM_FP.integrate = 0

EM_FP.probe_geometry = Line

EM_FP.x_probe = 0
EM_FP.y_probe = 0
EM_FP.z_probe = -55.e-6

EM_FP.x1_probe = 0
EM_FP.y1_probe = 0
EM_FP.z1_probe = 41.e-6
EM_FP.resolution = 500
EM_FP.do_moving_window_FP = 1
```

**Comment by ax3l** [comment]
Hank you for reporting this. What is the physical extend of your simulation in:
```
...
geometry.dims = 3
geometry.prob_lo = ...
geometry.prob_hi = ...
...
```

and what moving window or boosted frame options do you set?

Are there any warnings showing up when you run the simulation?

What is the latest version of WarpX you tried, `23.12`?

**Comment by sheroytata** [comment]
Thanks for the reply, sorry I seem to have missed this message since there was no email sent to me after your response.

The extents of the simulation are
geometry.prob_lo     = -80.e-6   -80.e-6   -55.e-6    # physical domain
geometry.prob_hi     =  80.e-6    80.e-6    41.e-6

its a moving window with in the z direction without a boosted frame.

There are no warnings  as the simulation begins.

No i have not tried the field probe with 23.12. 
I noticed that the conda builds (23.12) seem to have some upper limit on the number of GPU's to 25.

Is it also possible to have updates to the spack repository as new releases are released?

Thanks,
Seems I ask for a lot.

**Reply by Poto45** [reply]
Hello,

I seem to be running into the same issue. Have you found a way for it to work? 

I'm working with a moving window of dimensions:
_geometry.prob_lo     = -0.00156     -0.0045   -0.0336
geometry.prob_hi     =  0.00156     0.002     -0.0076_

With the window center going from around -0.02 to about 0.1. I've tried point and line diags and the following are the commands in the WarpX input script. 

_warpx.reduced_diags_names               = FP
FP.type = FieldProbe
FP.intervals = 10
FP.integrate = 0
FP.probe_geometry = Point
FP.x_probe = 0
FP.y_probe = 0
FP.z_probe = .01
#FP.x1_probe = 0 
#FP.y1_probe = 0
#FP.z1_probe = .04
#FP.resolution = 300
FP.do_moving_window_FP = 1_

I'm uploading the log and backtrace files I have, but I haven't been able to figure out what the error is. The simulation stops as soon as the moving window domain reaches the z point for the diagnostics. 
[backtracefile.txt](https://github.com/user-attachments/files/16666773/backtracefile.txt)
[logfile.txt](https://github.com/user-attachments/files/16666774/logfile.txt)

**Reply by ax3l** [reply]
@sheroytata Regarding your other questions:
>  I noticed that the conda builds (23.12) seem to have some upper limit on the number of GPU's to 25.

We currently do only release a CPU version of WarpX on conda, since the build system needs a lot of work to support GPU builds. But you can easily build WarpX using conda dependencies for GPU: https://warpx.readthedocs.io/en/latest/install/dependencies.html#conda-linux-macos-windows

> Is it also possible to have updates to the spack repository as new releases are released?

Good timing, I just updated our Spack packages this weekend https://github.com/spack/spack/pull/45251 and we will update them again monthly going forward. (We had a larger build system update on our Python end introducing pyAMReX that took more time to complete in Spack.)

**Reply by ax3l** [reply, marked as answer]
@Poto45 
> I seem to be running into the same issue. Have you found a way for it to work?

Your initial field probe  in `z=.01` is outside of the simulation [-0.0336 to  -0.0076], which is why the particle is removed. With `FP.do_moving_window_FP = 1`  if would be outside of the domain forever.

Note that we do not yet support field probes that enter a domain at a later time, but that would be a good feature to add: https://github.com/ECP-WarpX/WarpX/issues/5153

**Reply by ax3l** [reply]
@sheroytata your field probe `EM_FP.z_probe = -55.e-6` might be just too close to the lower boundary in z for your geometry ` [-55.e-6 to 41.e-6]` and gets remove as well. Try moving it to a slightly larger position initially?

**Reply by Poto45** [reply]
> @Poto45
> 
> > I seem to be running into the same issue. Have you found a way for it to work?
> 
> Your initial field probe in `z=.01` is outside of the simulation [-0.0336 to -0.0076], which is why the particle is removed. With `FP.do_moving_window_FP = 1` if would be outside of the domain forever.
> 
> Note that we do not yet support field probes that enter a domain at a later time, but that would be a good feature to add: #5153

Alright. I've gotten it to write data to the file now, if the point starts within the moving window domain (for both 0 and 1 of do_moving_window_FP). For both cases, though, I get the same information and only up to z=0.0256. Also, in both cases, the part_z_lev0- never changes from what I set it to as -0.0078, so it doesn't appear to be following the moving window. 

However, I really want the poynting vector as a specific location in the simulation that is at the end of the structure. This means that it won't be within the moving window domain when it starts. If I do the simulation with do_moving_window_FP=0 and have the z=0.01, for example, the simulation fails. Does that mean that this cannot be done? Thanks. 

(as an example, say I want the poynting vector along the dotted line in the photo attached)
![PXL_20240820_171204316](https://github.com/user-attachments/assets/db4b0621-8b21-4fe9-a3a1-3e32898c91ca)

**Comment by sheroytata** [comment]
Dear Axl,
Finally, I understood the field probe!

The intervals is the interval parser... FP.intervals=100 just does not
work! Its not documented as far as i saw (even in the examples as far as I
remember)

The one below works!
warpx.reduced_diags_names = FP
FP.type = FieldProbe
FP.intervals = 1:200:2

Sheroy
